claat/proto-renderer/generic_renderer_test.go

```diff
@@ -0,0 +1,60 @@
+package genrenderer
+
+import (
+	"go/build"
+	"path/filepath"
+	"testing"
+	"text/template"
+)
+
+type encapsulatedTest struct {
+	in  sampleProtoTemplate
+	out string
+	ok  bool
+}
+
+type sampleProtoTemplate struct {
+	Value interface{}
+}
+
+func newSampleProtoTemplate(el interface{}) sampleProtoTemplate {
+	return sampleProtoTemplate{Value: el}
+}
+
+func TestExecuteTemplate(t *testing.T) {
+	tmplsRltvDir := "src/github.com/googlecodelabs/tools/claat/proto-renderer/testdata/*"
+	tmplsAbsDir := filepath.Join(build.Default.GOPATH, tmplsRltvDir)
+	funcMap := template.FuncMap{
+		"returnInt": func(i int) int { return i },
+	}
+	tmpl := template.Must(template.New("dummy").Funcs(funcMap).ParseGlob(tmplsAbsDir))
+
+	tests := []encapsulatedTest{
+		{newSampleProtoTemplate(3), "3", false},
+		{newSampleProtoTemplate(nil), "", true},
+		{newSampleProtoTemplate("not-valid"), "", true},
+	}
+
+	for _, tc := range tests {
+		runEncapsulatedTest(tc, tmpl, t)
+	}
+}
+
+func runEncapsulatedTest(test encapsulatedTest, tmpl *template.Template, t *testing.T) {
```

Comment:
This can be an inline/nested func instead of a helper, and still works. helper functions seem fine?